Refactor api endpoints

diraulo opened the pull request:
- Make use of `current_user` in `order` controller instead of passing `user_id` with request
- Move `menu_id` needed to update inventory to `order_items` 

TODO: not update stock inventory when `order` is not created for some reason.

diraulo committed:
Remove menu_id from order and add to order_items

Having `menu_id` at the order, makes it impossible to create
an order with items from a different menu

diraulo committed:
Remove user_id from session response and use current_user in order controller

razorcd commented on app/controllers/api/v1/orders_controller.rb:
```diff
@@ -12,9 +13,9 @@ class Api::V1::OrdersController < ApiController
 
   def create
-    attributes = order_params
+    attributes = order_params.merge(user_id: current_user.id) if current_user
```
is `if current_user` necessary here? 
looking at `authenticate_api_user` method, it is instantly returning 403 if no user is found. We should always have a `current_user` at this point.

diraulo replied:
Ok you're right, I didn't check the implementation of that method :sweat:

razorcd commented on app/controllers/api/v1/orders_controller.rb:
```diff
@@ -38,33 +39,30 @@ def convert_json_to_params
 
   def order_params
-    @json_params.require(:order).permit(:user_id, :status, :order_time, :pickup_time, :menu_items)
+    @json_params.require(:order).permit(:status, :order_time, :pickup_time)
```
should we use `params.require...` here instead of `@json_params.require...` ? 
Because we have `before_action :convert_json_to_params, only: [:create, :update]`

diraulo committed:
Merge remote-tracking branch 'upstream/develop' into refactor-api-endpoints

diraulo committed:
Refactor order controller

- Return a better response when order can't be created
- Extract logic to add `order_items` to `order` to it's own method
  called `process_order_items`

diraulo commented:
I'm done with this PR, if there's no issues i think we can merge it
